.github/workflows/deploy-mainnet.yml

```diff
@@ -2,6 +2,8 @@ name: Deploy to Mainnet network
 
 # Run on pushes to master
 on:
+  # Pull request hook without any config. Launches for every pull request
+  pull_request:
```

Comment:
WDYT about making the builds based on [`tags`](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#example-including-branches-and-tags)? I mean, not all the PRs require to be built for `mainnet` isn't it?